src/vs/workbench/services/editor/common/customEditorLabelService.ts

```diff
@@ -39,6 +40,8 @@ export class CustomEditorLabelService extends Disposable implements ICustomEdito
 	private patterns: ICustomEditorLabelPattern[] = [];
 	private enabled = true;
 
+	private cache = new MRUCache<URI, string>(1000);
```

Comment:
Even though the URI reference would be more performant as the key, I think we should use the unique identifier of an URI as it is possible the `getName()` method will be called from different places for the same file. This means the key would be a `string`. 
The `URI` class has a `toString()` method which creates a unique identifier of the resource, use this method to get the key.